openlibrary/plugins/openlibrary/code.py

```diff
@@ -787,6 +787,8 @@ def is_bot():
         'buzzbot', 'laserlikebot', 'baiduspider', 'bingbot',
         'mj12bot', 'yoozbotadsbot'
     ]
+    if 'HTTP_USER_AGENT' not in web.ctx.env:
```

Comment:
What if `HTTP_USER_AGENT` *is* in `web.ctx.env` but is `None`?

I think it may be advantageous for us to instead say:

```
if not web.ctx.env.get('HTTP_USER_AGENT'):
    return True
```

thoughts?

Reply:
Should always exist, may be None

Reply:
Keyword being *should* :). I'd still go with the code snippet above:

```
if not web.ctx.env.get('HTTP_USER_AGENT'):
    return True
```